### ISSUE TITLE
TFLite Interpreter crashes with null TfLiteDelegatePtr

### ISSUE THREAD

**Issue by kalaluthien** [tensorflow/tensorflow#45191]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): r2.3 but consistent with latter versions
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
`tflite::Interpreter` does not check nullity of `TfLiteDelegate*`.
It would be better to check it and return ErrorCode for `null`.

https://github.com/tensorflow/tensorflow/blob/595d575c237368c925229c05e933d62808330ca2/tensorflow/lite/core/subgraph.cc#L1486-L1513

**Will this change the current api? How?**

https://github.com/tensorflow/tensorflow/blob/595d575c237368c925229c05e933d62808330ca2/tensorflow/lite/core/subgraph.cc#L1486

or

https://github.com/tensorflow/tensorflow/blob/595d575c237368c925229c05e933d62808330ca2/tensorflow/lite/interpreter.cc#L385

or (for r2.3 I am using)

https://github.com/tensorflow/tensorflow/blob/r2.3/tensorflow/lite/interpreter.cc#L358

returns `TfLiteError` for `null` input.

**Who will benefit with this feature?**

App developers who are using TensorflowLite SDK would find bugs more quickly.
Thanks!

**Any Other info.**

**Comment by srjoglekar246** [comment]
Hmm. This should be the behavior, yes. However, I may not get to this very soon, since such behavior will probably only occur if the user is using a non-recommended way of initializing delegates.

Out of curiosity, how did you encounter a seg-fault because of an invalid `TfLiteDelegate*`? Our language bindings usually initialize delegates in a safe way. For C++, the recommended ways of initializing delegates should avoid this using Create/DeleteDelegate methods (found in individual delegate documentation) OR using `std::unique_ptr` like [here](https://github.com/tensorflow/tensorflow/blob/595d575c237368c925229c05e933d62808330ca2/tensorflow/lite/tools/evaluation/utils.h#L79).

Feel free to send a PR :-).

**Comment by kalaluthien** [comment]
Thanks for your reply 👍 
I agree with you about the impact of this change. I'll send a small PR.

Answer: I found the issue when dealing with cross-platform code similar to below pseudo code.
It was hard to find because it works for most of the cases!
(I wrote macros just for giving intuitions here)

```cpp
...
using TfLiteDelegatePtr = std::unique_ptr<TfLiteDelegate, void (*)(TfLiteDelegate*)>;

TfLiteDelegatePtr AcquireDelegate() {
#if defined(OUTPUT_LINKED_WITH_OPENCL_OR_OPENGL)
  ...
  TfLiteDelegatePtr delegate(TfLiteGpuDelegateV2Create(&opts), TfLiteGpuDelegateV2Delete);
#elif defined(OUTPUT_LINKED_WITH_METAL)
  ...
  TfLiteDelegatePtr delegate(TFLGpuDelegateCreate(&opts), TFLGpuDelegateDelete);
#elif defined(OUTPUT_LINKED_WITH_XNNPACK)
  ...
  TfLiteDelegatePtr delegate(TfLiteXNNPackDelegateCreate(&opts), TfLiteXNNPackDelegateDelete);
#else  // OUTPUT_CANNOT_BE_LINKED_BECAUSE_THE_TARGET_PLATFORM_SUPPORTS_NOTHING
  /* Cannot use any delegate-related symbols that may require other dependency. */
  /* Of course to use `DummyDelegate` with default option may be more elegant way to do this. */
  TfLiteDelegatePtr delegate(nullptr, [](TfLiteDelegate*) {});
#endif
  return delegate;
}
```

**Comment by kalaluthien** [comment]
I think this is solved at d22d31b
